doc/sphinx/language/core/assumptions.rst

```diff
@@ -71,7 +71,21 @@ the list of binders, it is expanded to a let-in definition (see
 Section :ref:`let-in`).
 
 The expression :n:`forall @ident : @type, @term` denotes the
-*product* of the variable :n:`@ident` of type :n:`@type`, over the term :n:`@term`.
+:gdef:`product type <product>` (or *product*) of the variable :n:`@ident` of type :n:`@type`,
+over the term :n:`@term`.  If :n:`@ident` is used in :n:`@term`, then we say
+the expression is a :gdef:`dependent product`.
+
+These terms are also useful:
+
+* `n` is a :gdef:`dependent premise` of `forall n:nat, n + 0 = n` because
```

Comment:
We discussed to change this into:
```suggestion
* `n:nat` is a :gdef:`dependent premise` of `forall n:nat, n + 0 = n` because
```

to be less inconsistent with the other case (where it is the type that we designate as the premise).